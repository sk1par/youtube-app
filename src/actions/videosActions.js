import youTubeApi from '../api/youtube';

export const setVideos = (videos) => ({
    type: 'ALL_VIDEOS',
    payload: videos
});

const videos = (term) => async (dispatch) => {
    const response = await youTubeApi.get('/search', {
        params: {
            q: term
        }
    })
    console.log(response);
    const videos = response.data.items;
    dispatch(setVideos(videos));
}

export default videos;