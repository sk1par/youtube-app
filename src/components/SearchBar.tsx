import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { videosActions } from '../actions';


const SearchBar = () => {
    const dispatch = useDispatch();
    const [term, setTerm] = useState<string>('');

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setTerm(e.target.value);
    };

    const handleSubmit = async (event: React.FormEvent) => {
        event.preventDefault();
        
        if (term) {
            dispatch(videosActions(term));
        }
    };

    return (
        <>
            <h2  style={{textAlign:"center"}}><img style={{width:'200px', height:'100px',justifyContent:'center'}} src='https://www.thatitguy.net/wp-content/uploads/2018/08/1280px-Logo_of_YouTube_2015-2017.svg.png' alt="youtube logo"></img></h2>
            <div className='search-bar ui segment' data-auto-id="search-div">
                <form onSubmit={handleSubmit} className='ui form'>
                    <div className='field'>
                        <label htmlFor="video-search">Video Search</label>
                        <input onChange={handleChange} name='video-search' type="text" placeholder="Search.." data-auto-id='input-field' />
                    </div>
                    <div className='button'>
                <button onClick={handleSubmit} className="ui black basic button" data-auto-id="submit-button">Search</button>
                </div>
                </form>
            </div>
        </>
    )
}

export default SearchBar
