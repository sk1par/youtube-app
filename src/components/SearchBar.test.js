import { useSelector } from 'react-redux';
import SearchBar from './SearchBar';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';
import videosActions from '../actions/videosActions';

Enzyme.configure({ adapter: new Adapter() });
jest.mock('react-redux', () => ({
    useDispatch: () => (fn) => () => fn(),
    useSelector: jest.fn(() => []),
}));

jest.mock('../actions/videosActions', () => {
    const videosActions = jest.fn();
    return videosActions;
});

jest.mock("../api/youtube", () => {
    const videos = [ getVideo() ];
    return {
      get: jest.fn(() => Promise.resolve(videos))
    };
  });

const getVideo = () => ({
    id: {
        videoId: '1',
    },
    snippet: {
        description: 'test',
        title: 'test',
        thumbnails: {
            medium: {
                url: 'test',
            },
        },
    },
});


it('should render correctly search div', () => {
    const wrapper = shallow(<SearchBar />);
    const searchDiv = wrapper.find('[data-auto-id="search-div"]');
    expect(searchDiv).toHaveLength(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should trigger videos action correctly after submit', () => {
    const wrapper = mount(<SearchBar />);
    const inputText = wrapper.find('[data-auto-id="input-field"]');
    inputText.simulate('change', { target: { value: 'ruse' } });
    const searchButton = wrapper.find('[data-auto-id="submit-button"]');
    searchButton.simulate('click');
    expect(videosActions).toHaveBeenCalledWith('ruse');
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

