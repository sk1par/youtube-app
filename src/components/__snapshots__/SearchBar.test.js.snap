// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly search div 1`] = `
<Fragment>
  <h2
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <img
      alt="youtube logo"
      src="https://www.thatitguy.net/wp-content/uploads/2018/08/1280px-Logo_of_YouTube_2015-2017.svg.png"
      style={
        Object {
          "height": "100px",
          "justifyContent": "center",
          "width": "200px",
        }
      }
    />
  </h2>
  <div
    className="search-bar ui segment"
    data-auto-id="search-div"
  >
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          htmlFor="video-search"
        >
          Video Search
        </label>
        <input
          data-auto-id="input-field"
          name="video-search"
          onChange={[Function]}
          placeholder="Search.."
          type="text"
        />
      </div>
      <div
        className="button"
      >
        <button
          className="ui black basic button"
          data-auto-id="submit-button"
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </form>
  </div>
</Fragment>
`;

exports[`should trigger videos action correctly after submit 1`] = `
<SearchBar>
  <h2
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <img
      alt="youtube logo"
      src="https://www.thatitguy.net/wp-content/uploads/2018/08/1280px-Logo_of_YouTube_2015-2017.svg.png"
      style={
        Object {
          "height": "100px",
          "justifyContent": "center",
          "width": "200px",
        }
      }
    />
  </h2>
  <div
    className="search-bar ui segment"
    data-auto-id="search-div"
  >
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          htmlFor="video-search"
        >
          Video Search
        </label>
        <input
          data-auto-id="input-field"
          name="video-search"
          onChange={[Function]}
          placeholder="Search.."
          type="text"
        />
      </div>
      <div
        className="button"
      >
        <button
          className="ui black basic button"
          data-auto-id="submit-button"
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </form>
  </div>
</SearchBar>
`;
